src/libs/agent-runtime/bedrock/index.ts

```diff
@@ -38,35 +46,48 @@ export class LobeBedrockAI implements LobeRuntimeAI {
     });
   }
 
-  async chat(payload: ChatStreamPayload) {
+  async chat(payload: ChatStreamPayload, options?: ChatCompetitionOptions) {
     if (payload.model.startsWith('meta')) return this.invokeLlamaModel(payload);
 
-    return this.invokeClaudeModel(payload);
+    return this.invokeClaudeModel(payload, options);
   }
 
   private invokeClaudeModel = async (
     payload: ChatStreamPayload,
+    options?: ChatCompetitionOptions
   ): Promise<StreamingTextResponse> => {
+    const { max_tokens, messages, model, temperature, top_p } = payload;
+    const system_message = messages.find((m) => m.role === 'system');
+    const user_messages = messages.filter((m) => m.role !== 'system');
+
     const command = new InvokeModelWithResponseStreamCommand({
       accept: 'application/json',
       body: JSON.stringify({
-        max_tokens_to_sample: payload.max_tokens || 400,
-        prompt: experimental_buildAnthropicPrompt(payload.messages as any),
+        anthropic_version: "bedrock-2023-05-31",
+        max_tokens: max_tokens || 4096,
+        messages: buildAnthropicMessages(user_messages),
+        system: system_message?.content as string,
+        temperature: temperature,
+        top_p: top_p,
       }),
       contentType: 'application/json',
-      modelId: payload.model,
+      modelId: model,
     });
 
     try {
       // Ask Claude for a streaming chat completion given the prompt
       const bedrockResponse = await this.client.send(command);
 
       // Convert the response into a friendly text-stream
-      const stream = AWSBedrockAnthropicStream(bedrockResponse);
+      const stream = AWSBedrockStream(bedrockResponse, options?.callback, (chunk) => {
+        if (chunk.type === 'content_block_delta') {
```

Comment:
Why need this?

Reply:
Ya, this part is probably not needed according to the official doc https://docs.anthropic.com/claude/reference/messages-streaming

Reply:
Removed this useless condition check